### PR TITLE
To add steps to support TrendMicro newer api setup

### PR DESCRIPTION
**Pull request by justjais**
To add steps to support TrendMicro newer api setup

### CODE REVIEW

File: playbooks/ansible-security-trendmicro-deepsec-appliance/files/bootstrap.yaml [hunk 1]
```diff
@@ -1,5 +1,79 @@
 ---
-- hosts: controller
-  gather_facts: false
-  connection: network_cli
-  tasks: []
+- name: Get API key for working with TM newer rest APIs
+  hosts: appliance
```
Comment (pabelanger) [inline playbooks/ansible-security-trendmicro-deepsec-appliance/files/bootstrap.yaml:3]: appliance*

Comment (pabelanger) [reply playbooks/ansible-security-trendmicro-deepsec-appliance/files/bootstrap.yaml:3]: or appliance-ssh

**Comment by pabelanger** [comment]
recheck

**Comment by justjais** [comment]
recheck

File: playbooks/ansible-test-network-integration-base/files/bootstrap-trendmicro.deepsec.deepsec.yaml [hunk 2]
```diff
@@ -16,3 +16,85 @@
...
+            cookie: "{{ request_id['set_cookie'] }}"
+            rID: "{{ request_id['json']['RID'] }}"
+            api-version: v1
+        register: api_key_response
+
+      - name: Create temp file with API Key
+        copy:
+          dest: "/tmp/deepsec_api_key.txt"
+          content: "{{ api_key_response['json']['secretKey'] }}"
+        delegate_to: 127.0.0.1
```
Comment (pabelanger) [inline playbooks/ansible-test-network-integration-base/files/bootstrap-trendmicro.deepsec.deepsec.yaml:67]: I don't think this is working as expected, you might want to, at the bottem. Create a new play, for the controller and write the file

File: playbooks/ansible-test-network-integration-base/files/bootstrap-trendmicro.deepsec.deepsec.yaml [hunk 3]
```diff
@@ -16,3 +16,89 @@
...
+
+      - name: ensure the API key file is created over controller
+        debug:
+          msg: "{{ lookup('file', '/tmp/deepsec_api_key.txt') }}"
```
Comment (pabelanger) [inline playbooks/ansible-test-network-integration-base/files/bootstrap-trendmicro.deepsec.deepsec.yaml:71]: this runs on the appliance, so if the file exists then the task before didn't work

File: roles/ansible-test-inventory/templates/inventory-trendmicro.deepsec.deepsec.j2 [hunk 4]
```diff
@@ -6,9 +6,8 @@ Trendmicro-DeepSecurity-20.0.344 ansible_host={{ hostvars['Trendmicro-DeepSecuri
 
 [deepsec:vars]
 ansible_network_os=trendmicro.deepsec.deepsec
-ansible_user=ec2-user
-ansible_httpapi_password=zuulzuul56
 ansible_httpapi_port=443
 ansible_httpapi_use_ssl=true
 ansible_httpapi_validate_certs=false
 ansible_connection=httpapi
+ansible_httpapi_session_key={'api-secret-key':"{{ lookup('file', '/tmp/deepsec_api_key.txt') }}", 'api-version': 'v1'}
```
Comment (pabelanger) [inline roles/ansible-test-inventory/templates/inventory-trendmicro.deepsec.deepsec.j2:13]: try
```
ansible_httpapi_session_key={{ lookup('file', '/tmp/deepsec_api_key.txt') }}
```

File: playbooks/ansible-security-trendmicro-deepsec-appliance/roles/tm-deepsec-aws-setup/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,204 @@
+---
+# tasks file for tm_silent_setup
+
+- name: Ensure bash, OpenSSl, and libssl are the latest versions
+  become: true
+  yum:
```
Comment (pabelanger) [inline playbooks/ansible-security-trendmicro-deepsec-appliance/roles/tm-deepsec-aws-setup/tasks/main.yml:6]: lets use package

File: playbooks/ansible-security-trendmicro-deepsec-appliance/roles/tm-deepsec-aws-setup/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+  become: true
+  file:
+    path: "{{ item }}"
+    mode: u=rwX,g=rX,o=rX
+    recurse: true
+  with_items: "{{ dir_permission }}"
+
+- name: Remove the postgresql package
+  become: true
+  yum:
```
Comment (pabelanger) [inline playbooks/ansible-security-trendmicro-deepsec-appliance/roles/tm-deepsec-aws-setup/tasks/main.yml:26]: package

File: playbooks/ansible-security-trendmicro-deepsec-appliance/roles/tm-deepsec-aws-setup/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+- name: Remove pgsql directory
+  become: true
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items: "{{ remove_dir }}"
+
+- name: Install postgres 11 libs rpm from a remote file
+  become: true
+  yum:
```
Comment (pabelanger) [inline playbooks/ansible-security-trendmicro-deepsec-appliance/roles/tm-deepsec-aws-setup/tasks/main.yml:40]: package

File: playbooks/ansible-security-trendmicro-deepsec-appliance/roles/tm-deepsec-aws-setup/tasks/main.yml (hunk 5 above)
Comment (pabelanger) [reply playbooks/ansible-security-trendmicro-deepsec-appliance/roles/tm-deepsec-aws-setup/tasks/main.yml:6]: these package tasks, could be moved into single instances with packages as a list

File: playbooks/ansible-test-network-integration-base/pre.yaml [hunk 8]
```diff
@@ -23,6 +23,7 @@
         tox_extra_args: "-vv -- ansible-playbook -v playbooks/ansible-test-network-integration-base/files/bootstrap-{{ _network_os }}.yaml"
         tox_install_siblings: false
         zuul_work_dir: "{{ zuul.projects['github.com/ansible/ansible-zuul-jobs'].src_dir }}"
+        trendmicro_deepsec_api_key: '{{ zuul.executor.work_root }}'
```
Comment (pabelanger) [inline playbooks/ansible-test-network-integration-base/pre.yaml:26]: what is this used for?

File: playbooks/ansible-test-network-integration-base/pre.yaml [hunk 9]
```diff
@@ -36,6 +37,7 @@
         ansible_test_inventory_os: "{{ _network_os }}"
         ansible_test_inventory_dest: ~/inventory
         vmware_ci_set_passwords_secret_dir: '{{ zuul.executor.work_root }}'
+        trendmicro_deepsec_api_key: '{{ zuul.executor.work_root }}'
```
Comment (pabelanger) [inline playbooks/ansible-test-network-integration-base/pre.yaml:40]: same comment, seems unused

Comment (goneri) [reply playbooks/ansible-test-network-integration-base/pre.yaml:40]: Yes, the path is hardcoded in roles/ansible-test-inventory/templates/inventory-trendmicro.deepsec.deepsec.j2.

I suggest to rename this variable from `trendmicro_deepsec_api_key` to `trendmicro_deepsec_secret_dir` to be consistent with wmare. And off course, reuse the variable in `inventory-trendmicro.deepsec.deepsec.j2`.

Comment (goneri) [reply playbooks/ansible-test-network-integration-base/pre.yaml:40]: I would actually prefer to use one single variable called `ansible_test_inventory_secret_dir`.

File: roles/ansible-test-inventory/templates/inventory-trendmicro.deepsec.deepsec.j2 [hunk 10]
```diff
@@ -6,9 +6,8 @@ Trendmicro-DeepSecurity-20.0.344 ansible_host={{ hostvars['Trendmicro-DeepSecuri
 
 [deepsec:vars]
 ansible_network_os=trendmicro.deepsec.deepsec
-ansible_user=ec2-user
-ansible_httpapi_password=zuulzuul56
 ansible_httpapi_port=443
 ansible_httpapi_use_ssl=true
 ansible_httpapi_validate_certs=false
 ansible_connection=httpapi
+ansible_httpapi_session_key={'api-secret-key': '{{ lookup('file', '{{ zuul.executor.work_root }}/Trendmicro-DeepSecurity-20.0.344/tmp/deepsec_api_key.txt') }}', 'api-version': 'v1'}
```
Comment (pabelanger) [inline roles/ansible-test-inventory/templates/inventory-trendmicro.deepsec.deepsec.j2:13]: @goneri trendmicro_deepsec_api_key isn't referenced here. Given we don't need it any other place, we can just hardcoded it?

File: zuul.d/ansible-security-jobs.yaml [hunk 11]
```diff
@@ -93,9 +93,9 @@
     parent: ansible-network-appliance-base
     pre-run: playbooks/ansible-security-trendmicro-deepsec-appliance/pre.yaml
     run: playbooks/ansible-security-trendmicro-deepsec-appliance/run.yaml
+    attempts: 1
```
Comment (pabelanger) [inline zuul.d/ansible-security-jobs.yaml:96]: this needs to be removed

File: playbooks/ansible-security-trendmicro-deepsec-appliance/pre.yaml [hunk 12]
```diff
@@ -1,15 +1,9 @@
 ---
-- hosts: controller
+- hosts: appliance*
+  gather_facts: false
+  vars:
+    trendmicro_deepsec_api_key: '{{ zuul.executor.work_root }}'
```
Comment (pabelanger) [inline playbooks/ansible-security-trendmicro-deepsec-appliance/pre.yaml:5]: unused can be removed

File: playbooks/ansible-security-trendmicro-deepsec-appliance/pre.yaml [hunk 13]
```diff
@@ -1,15 +1,9 @@
...
   tasks:
-    - name: Ensure tox
+    - name: Setup TM instance silently
       include_role:
-        name: ensure-tox
```
Comment (pabelanger) [inline playbooks/ansible-security-trendmicro-deepsec-appliance/pre.yaml:6]: for now, revert this. and in the follow up PR we can remove and address ARA failures

**Comment by pabelanger** [comment]
recheck

**Pull request merged**